MDANSE/Src/MDANSE/Framework/Jobs/NeutronDynamicTotalStructureFactor.py

```diff
@@ -394,58 +400,33 @@ def finalize(self):
             bj = self.configuration["trajectory"]["instance"].get_atom_property(
                 pair[1], "b_coherent"
             )
-            ni = nAtomsPerElement[pair[0]]
-            nj = nAtomsPerElement[pair[1]]
-            ci = ni / nTotalAtoms
-            cj = nj / nTotalAtoms
-
-            self._outputData[f"f(q,t)_coh_weighted_{pair_str}"][:] = (
-                self._outputData[f"f(q,t)_coh_{pair_str}"][:]
-                * np.sqrt(ci * cj)
-                * bi
-                * bj
-            )
-            self._outputData[f"s(q,f)_coh_weighted_{pair_str}"][:] = (
-                self._outputData[f"s(q,f)_coh_{pair_str}"][:]
-                * np.sqrt(ci * cj)
-                * bi
-                * bj
-            )
+
             if pair[0] == pair[1]:  # Add a factor 2 if the two elements are different
-                self._outputData["f(q,t)_coh_total"][:] += self._outputData[
-                    f"f(q,t)_coh_weighted_{pair_str}"
-                ][:]
-                self._outputData["s(q,f)_coh_total"][:] += self._outputData[
-                    f"s(q,f)_coh_weighted_{pair_str}"
-                ][:]
+                self._outputData[f"f(q,t)_coh_{pair_str}"] *= bi * bj * norm_natoms
+                self._outputData[f"s(q,f)_coh_{pair_str}"] *= bi * bj * norm_natoms
             else:
-                self._outputData["f(q,t)_coh_total"][:] += (
-                    2 * self._outputData[f"f(q,t)_coh_weighted_{pair_str}"][:]
-                )
-                self._outputData["s(q,f)_coh_total"][:] += (
-                    2 * self._outputData[f"s(q,f)_coh_weighted_{pair_str}"][:]
-                )
+                self._outputData[f"f(q,t)_coh_{pair_str}"] *= 2 * bi * bj * norm_natoms
+                self._outputData[f"s(q,f)_coh_{pair_str}"] *= 2 * bi * bj * norm_natoms
+
+            self._outputData["f(q,t)_coh_total"][:] += self._outputData[
+                f"f(q,t)_coh_{pair_str}"
+            ][:]
+            self._outputData["s(q,f)_coh_total"][:] += self._outputData[
+                f"s(q,f)_coh_{pair_str}"
+            ][:]
 
         # Compute incoherent functions and structure factor
-        for element, ni in nAtomsPerElement.items():
+        for element in nAtomsPerElement.keys():
```

Comment:
If this is a normal dict, `keys` is default.
```suggestion
        for element in nAtomsPerElement:
```